types/submessages.go

```diff
@@ -81,6 +81,12 @@ type SubMsgResult struct {
 // with full Cosmos SDK events.
 // This mirrors Rust's SubMsgResponse.
 type SubMsgResponse struct {
-	Events Events `json:"events"`
-	Data   []byte `json:"data,omitempty"`
+	Events       Events        `json:"events"`
+	Data         []byte        `json:"data,omitempty"`
+	MsgResponses []MsgResponse `json:"msg_responses"`
```

Comment:
For `Events` we have an extra type to customize the JSON. But I wonder if this is needed here. What's the default value of []MsgResponse? Nil or the empty slice? Any thoughts?

Reply:
Oh, that's a really good catch! The default value in Go is empty slice, but when serializing to JSON, it becomes `null`. So, we should add a similar type here.
I also think we need to add something like this to the `go-gen`. It works fine in the direction Rust -> Go, but for Go -> Rust the code is not correct.

Reply:
Some idea, not sure: I heard Go can do generics now. Is this something an `Array<T>` could solve and we use `Array<Event>` and `Array<MsgResponse>` here?

Reply:
Nice, I was thinking the same. Yes, that should work